src/WebJobs.Extensions.ApiHub/Config/ApiHubConfiguration.cs

```diff
@@ -154,7 +154,25 @@ private Task<ApiHubFile> BuildFromAttribute(ApiHubFileAttribute attribute)
 
         private IFolderItem GetFileSource(string key)
         {
-            return _map[key];
+            if (_map.ContainsKey(key))
+            {
+                return _map[key];
+            }
+            else
+            {
+                // If the key doesn't exist, see if it is available as an environment variable.
```

Comment:
Basically pass the Binder object and do the binding inside the method/function itself.
Found the issue by this post:
http://stackoverflow.com/questions/43348985/dynamic-output-file-name-of-apihubfile-azure-function-binding-one-drive-drop-b